packages/graphqlgen/src/generators/common.ts

```diff
@@ -239,9 +239,10 @@ function deepResolveInputTypes(
     const childTypes = type.fields
       .filter(t => t.type.isInput && !seen[t.type.name])
       .map(t => t.type.name)
-      .map(name =>
-        deepResolveInputTypes(inputTypesMap, name, { ...seen, [name]: true }),
-      )
+      .map(name => {
+        seen[name] = true
```

Comment:
What is the intended effect of your change, given:

```ts
import * as _ from 'lodash'
const seen = {}
const name = 'bar'

const seenCopy = { ...seen, [name]: true }
seen[name] = true

assert(_.isEqual(seen, seenCopy))


```

?

Reply:
Ah, but by using mutation to share state across the recursion, we effectively guarantee that sibling traversal branches share their `seen` state...